service/src/MSRIOGroup.cpp

```diff
@@ -1193,25 +1190,23 @@ namespace geopm
     std::string MSRIOGroup::platform_data(int cpu_id)
     {
         std::string platform_msrs;
-        switch (cpu_id) {
-            case MSRIOGroup::M_CPUID_KNL:
-                platform_msrs = knl_msr_json();
-                break;
-            case MSRIOGroup::M_CPUID_HSX:
-            case MSRIOGroup::M_CPUID_BDX:
-                platform_msrs = hsx_msr_json();
-                break;
-            case MSRIOGroup::M_CPUID_SNB:
-            case MSRIOGroup::M_CPUID_IVT:
-                platform_msrs = snb_msr_json();
-                break;
-            case MSRIOGroup::M_CPUID_SKX:
-            case MSRIOGroup::M_CPUID_ICX:
-                platform_msrs = skx_msr_json();
-                break;
-            default:
-                throw Exception("MSRIOGroup: Unsupported CPUID",
-                                GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+        if (cpu_id == MSRIOGroup::M_CPUID_KNL) {
+            platform_msrs = knl_msr_json();
+        }
+        else if (cpu_id == MSRIOGroup::M_CPUID_HSX ||
+                 cpu_id == MSRIOGroup::M_CPUID_BDX) {
+            platform_msrs = hsx_msr_json();
+        }
+        else if (cpu_id == MSRIOGroup::M_CPUID_SNB ||
+                 cpu_id == MSRIOGroup::M_CPUID_IVT) {
+            platform_msrs = snb_msr_json();
+        }
+        else if (cpu_id >= MSRIOGroup::M_CPUID_SKX) {
```

Comment:
Do we want to add an ifdef GEOPM_DEBUG warning if cpu_id > M_CPUID_SKX that indicates we're using the skx msr because a newer MSR json isn't specified?

Reply:
I agree here.  We should maintain a list of known cpu_id values that are tested, and print a debug warning if we find a newer cpu_id value than the ones we have validated.